static/js/components/EducationForm.js

```diff
@@ -30,29 +32,29 @@ class EducationForm extends ProfileFormFields {
     setEducationDegreeInclusions:   React.PropTypes.func,
   };
 
-  openEditEducationForm = index => {
+  openEditEducationForm: Function = (index: number): void => {
     openEditEducationForm.call(this, index);
   };
 
-  openNewEducationForm = (level, index) => {
+  openNewEducationForm: Function = (level: string, index: number): void => {
     openNewEducationForm.call(this, level, index);
   };
 
-  deleteEducationEntry = () => {
+  deleteEducationEntry: Function = (): void => {
     deleteEducationEntry.call(this);
   };
 
-  renderEducationLevel(level){
+  renderEducationLevel(level: {value: string; label: string}) {
     const {
       ui: { educationDegreeInclusions },
       profile: { education },
     } = this.props;
     if (educationDegreeInclusions[level.value]) {
-      let rows = [];
+      let rows: Array<React$Element|void> = [];
       if (education !== undefined) {
-        rows = Object.entries(education).
-          filter(([, education]) => education.degree_name === level.value).
-          map(([index, education]) => this.educationRow(education, index));
+        rows = education.map((entry, index) => (
```

Comment:
I think the problem stems from `profile` not being inferable. Flow doesn't use `propTypes`, we have to define it separately using `props: { ... }` on the class